src/iceberg/delete_file_index.cc

```diff
@@ -0,0 +1,769 @@
+      continue;
+    }
+
+    const int32_t spec_id = manifest.partition_spec_id;
+    auto spec_iter = specs_by_id_.find(spec_id);
+    ICEBERG_CHECK(spec_iter != specs_by_id_.cend(),
+                  "Partition spec ID {} not found when loading delete files", spec_id);
+
+    const auto& spec = spec_iter->second;
+
+    // Get or compute projected partition expression
+    if (!part_expr_cache.contains(spec_id) && data_filter_) {
+      auto projector = Projections::Inclusive(*spec, *schema_, case_sensitive_);
+      ICEBERG_ASSIGN_OR_RAISE(auto projected, projector->Project(data_filter_));
+      part_expr_cache[spec_id] = std::move(projected);
+    }
+
+    // Get or create manifest evaluator
+    if (!eval_cache.contains(spec_id)) {
+      auto filter = partition_filter_;
+      if (auto it = part_expr_cache.find(spec_id); it != part_expr_cache.cend()) {
+        if (filter) {
+          ICEBERG_ASSIGN_OR_RAISE(filter, And::Make(filter, it->second));
+        } else {
+          filter = it->second;
+        }
+      }
+      if (filter) {
+        ICEBERG_ASSIGN_OR_RAISE(auto evaluator,
+                                ManifestEvaluator::MakePartitionFilter(
+                                    std::move(filter), spec, *schema_, case_sensitive_));
+        eval_cache[spec_id] = std::move(evaluator);
+      }
+    }
+
+    // Evaluate manifest against filter
+    if (auto it = eval_cache.find(spec_id); it != eval_cache.end()) {
+      ICEBERG_ASSIGN_OR_RAISE(auto should_match, it->second->Evaluate(manifest));
+      if (!should_match) {
+        continue;  // Manifest doesn't match filter
+      }
+    }
+
+    // Read manifest entries
+    ICEBERG_ASSIGN_OR_RAISE(auto reader,
+                            ManifestReader::Make(manifest, io_, schema_, spec));
+
+    auto partition_filter = partition_filter_;
+    if (auto it = part_expr_cache.find(spec_id); it != part_expr_cache.cend()) {
+      if (partition_filter) {
+        ICEBERG_ASSIGN_OR_RAISE(partition_filter,
+                                And::Make(partition_filter, it->second));
+      } else {
+        partition_filter = it->second;
+      }
+    }
+    if (partition_filter) {
+      reader->FilterPartitions(std::move(partition_filter));
+    }
+    if (partition_set_) {
+      reader->FilterPartitions(partition_set_);
+    }
+    reader->FilterRows(data_filter).CaseSensitive(case_sensitive_).TryDropStats();
+
+    ICEBERG_ASSIGN_OR_RAISE(auto entries, reader->LiveEntries());
+    files.reserve(files.size() + entries.size());
+
+    for (auto& entry : entries) {
+      ICEBERG_CHECK(entry.data_file != nullptr, "ManifestEntry must have a data file");
+      ICEBERG_CHECK(entry.sequence_number.has_value(),
+                    "Missing sequence number for delete file: {}",
+                    entry.data_file->file_path);
+      if (entry.sequence_number.value() > min_sequence_number_) {
+        files.emplace_back(std::move(entry));
```

Comment:
Java keeps stats only for columns needed for filtering:
  - Position deletes: Only DELETE_FILE_PATH column
  - Equality deletes: Only columns in equalityFieldIds()